File: Práctica 1/EjerciciosRuby/Parcela.rb
# encoding:utf-8
module EjerciciosRuby
  class Parcela
    def initialize(nombre,precio_compra,precio_base_alquiler,precio_edificar,num_casas,num_hoteles)
      @nombre = nombre
      @precio_compra = precio_compra
      @precio_base_alquiler = precio_base_alquiler
      @precio_edificar = precio_edificar
      @num_casas = num_casas
      @num_hoteles = num_hoteles
    end

    def nombre
      @nombre
    end

    def precioCompra
      @precio_compra
    end
    def precioEdificar
      @precio_edificar
    end
    def numCasas
      @num_casas
    end
    def numHoteles
      @num_hoteles
    end
    def precio_alquiler_completo
      precio_alquiler = @precio_base_alquiler*(1+@num_casas+(@num_hoteles*4))
    end
    def construir_casa
      @num_casas = @num_casas + 1
    end
    def construir_hotel
      @num_hoteles = @num_hoteles + 1
    end
    def getPrecioAlquilerDescuento(descuento,porc)
      if(descuento == true)
        precio_alquiler = @precio_base_alquiler*(1+@num_casas+(@num_hoteles*4))*(1-porc)
      end
      if(descuento == false)
        precio_alquiler = @precio_base_alquiler*(1+@num_casas+(@num_hoteles*4))
      end
    end
  sitio = Parcela.new('Juan Pablo',3,150,5,1,1)
  puts(sitio.precio_alquiler_completo)
  puts("Añadiendo hoteles...")
  sitio.construir_hotel
  puts(sitio.precio_alquiler_completo)
    descuento = false
    porc = 0.43
    puts(sitio.getPrecioAlquilerDescuento(descuento,porc))
  end
end

File: Práctica 1/EjerciciosJava/src/ejerciciosjava/Parcela.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejerciciosjava;

/**
 *@author Inés Jiménez
 * 2ºD D1
 */
public class Parcela {
    /**
     * Datos de la clase Parcela, son privados.
     */
    private String nombre;
    private float precioCompra,precioEdificar,precioBaseAlquiler;
    private int numCasas,numHoteles;
    /**
     * Constructor de la clase Parcela
    */
    public Parcela(String Nombre,float precio_compra,float precio_edificar,     
            float precio_base_alquiler, int num_casas, int num_hoteles){ 
        nombre = Nombre;
        precioCompra = precio_compra;
        precioEdificar = precio_edificar;
        precioBaseAlquiler = precio_base_alquiler;
        numCasas = num_casas;
        numHoteles = num_hoteles;
    }
    
    /**
     * métodos get:
     */
    public String getNombre(){
        return this.nombre;
    }
    public float getPrecioCompra(){
        return this.precioCompra;
    }
    public float getPrecioEdificar(){
        return this.precioEdificar;
    }
    public int getNumCasas(){
        return this.numCasas;
    }
    public int getNumHoteles(){
        return this.numHoteles;
    }
    /**
     * devuelve el precio del alquiler.
    */
    public double getPrecioAlquilerCompleto(){
        double precio = 0;
        precio = precioBaseAlquiler*(1+numCasas+numHoteles*4);
        return precio;
    }
    
    /**
     * sobrecarga del método getPrecioAlquilerCompleto()
     */
    public double getPrecioAlquilerCompleto(boolean descuento, int porcentaje){
        double precio = 0;
        double porc = porcentaje;
        if(descuento == true){
            precio = precioBaseAlquiler*(1+numCasas+numHoteles*4);
            precio = precio * (1-(porc/100));
        }
        else{
            precio = precioBaseAlquiler*(1+numCasas+numHoteles*4);
        }
        return precio;
    }
    /**
     * construir casas
     * @return true, siempre devuelve true
    */
    
    public boolean construirCasa(){
        numCasas++;
        return true;                                                           
    }
    
    /**
     * construir hoteles
     * @return true, siempre devuelve true
     */
    public boolean construirHotel(){
        numHoteles++;
        return true;                                                            
    }
    
    public boolean igualdadIdentidad (Parcela otraParcela){
        boolean identidad = false;
        if(otraParcela == this)
            identidad = true;
        return identidad;
    }
    
    public boolean igualdadEstado (Parcela otraParcela){
              boolean estado = false;
        if(otraParcela.nombre.equals(this.nombre)){
            if(otraParcela.numCasas == this.numCasas){
               if(otraParcela.numHoteles == this.numHoteles){
                   if(otraParcela.precioBaseAlquiler == this.precioBaseAlquiler){
                       if(otraParcela.precioCompra == this.precioCompra){
                           if(otraParcela.precioEdificar == this.precioEdificar){
                               estado = true;
                           }
                       }
                   }
               }
            }
        }
        return estado;
    }
}
